src/detect-bytemath.c

```diff
@@ -440,32 +451,28 @@ static void DetectByteMathFree(DetectEngineCtx *de_ctx, void *ptr)
  *
  * \retval A pointer to the SigMatch if found, otherwise NULL.
  */
-SigMatch *DetectByteMathRetrieveSMVar(const char *arg, int sm_list, const Signature *s)
+const SigMatch *DetectByteMathRetrieveSMVar(
+        const char *arg, int sm_list, int *found_list, const Signature *s)
 {
     for (uint32_t x = 0; x < s->init_data->buffer_index; x++) {
-        SigMatch *sm = s->init_data->buffers[x].head;
+        const SigMatch *sm = s->init_data->buffers[x].head;
         while (sm != NULL) {
-            if (sm->type == DETECT_BYTEMATH) {
-                const DetectByteMathData *bmd = (const DetectByteMathData *)sm->ctx;
-                if (strcmp(bmd->result, arg) == 0) {
-                    SCLogDebug("Retrieved SM for \"%s\"", arg);
-                    return sm;
-                }
+            if (DetectByteMathSMNameMatch(sm, arg)) {
```

Comment:
Still https://github.com/OISF/suricata/pull/13564#discussion_r2183560050

> This change looks suspicious

Did you test a rule with 2 buffers, each buffer doing a byte_extract+ byte_math ?

I am afraid that with this change, the second buffer will get their found_list from the first buffer when it uses to get the right sm_list == list in current code

Do you have a test for it ?

Reply:
Ok... I will test it